buildSrc/src/main/java/com/google/firebase/gradle/plugins/FireEscapeArtifactPlugin.kt

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.bundling.Zip
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.getByName
+import org.gradle.kotlin.dsl.register
+
+class FireEscapeArtifactPlugin : Plugin<Project> {
+
+  override fun apply(target: Project) {
+    target.afterEvaluate {
+      if (supportsMavenPublishing(target)) {
+        val apiTxtFile = registerApiTxtFileTask(project)
+        val proguardMappingFile = registerProguardMappingFileTask(project)
+        val javadoc = registerJavadocTask(project)
+
+        val zippedArtifact =
+          project.tasks.register<Zip>("maven") {
+            from(apiTxtFile)
+            if (project.isAndroid()) {
+              from(proguardMappingFile)
+            }
+          }
+
+        extensions.configure<PublishingExtension> {
+          publications.getByName<MavenPublication>("mavenAar") {
+            artifact(zippedArtifact)
+            artifact(javadoc)
+          }
+        }
+      }
+    }
+  }
+  private fun supportsMavenPublishing(project: Project): Boolean {
```

Comment:
Since these only have a single return statement, you can use `=` instead of `{}`

Reply:
I get compiler warnings about not specifying return type and end up having a more verbose lambda signature if I switch to this syntax, what's the cost/benefit going on with = functions or is it just an aesthetic choice?

Reply:
It's mostly to reduce boilerplate. What warnings are you getting?

Reply:
Curious about this too, because I don't get any warnings. Only if the return type is too ambiguous. Additionally, you can do this as well:
```kotlin
private fun supportsMavenPublishing(project: Project): Boolean =
  project.plugins.hasPlugin(MavenPublishPlugin::class.java)
```

Reply:
I tried again and got no problems, sometimes it's that easy I suppose